kotlinx-coroutines-core/jvm/test/lincheck/ChannelsLincheckTest.kt

```diff
@@ -15,107 +15,160 @@ import org.jetbrains.kotlinx.lincheck.*
         e.testResult
     }
 
-    @Operation
-    fun trySend(@Param(name = "value") value: Int): Any = c.trySend(value)
-            .onSuccess { return true }
-            .onFailure {
-                return if (it is NumberedCancellationException) it.testResult
-                else false
-            }
-
-    @Operation(promptCancellation = true)
-    suspend fun sendViaSelect(@Param(name = "value") value: Int): Any = try {
-        select<Unit> { c.onSend(value) {} }
-    } catch (e: NumberedCancellationException) {
-        e.testResult
-    }
+    // @Operation TODO: `trySend()` is not linearizable as it can fail due to postponed buffer expansion
+    //            TODO: or make a rendezvous with `tryReceive`, which violates the sequential specification.
+    open fun trySend(@Param(name = "value") value: Int): Any = c.trySend(value)
+        .onSuccess { return true }
+        .onFailure {
+            return if (it is NumberedCancellationException) it.testResult
+            else false
+        }
 
-    @Operation(promptCancellation = true)
+    @Operation(allowExtraSuspension = true, blocking = true)
     suspend fun receive(): Any = try {
         c.receive()
     } catch (e: NumberedCancellationException) {
         e.testResult
     }
 
-    @Operation
+    @Operation(allowExtraSuspension = true, blocking = true)
+    suspend fun receiveCatching(): Any = c.receiveCatching()
+        .onSuccess { return it }
+        .onClosed { e -> return (e as NumberedCancellationException).testResult }
+
+    @Operation(blocking = true)
     fun tryReceive(): Any? =
         c.tryReceive()
             .onSuccess { return it }
             .onFailure { return if (it is NumberedCancellationException) it.testResult else null }
 
-    @Operation(promptCancellation = true)
+    @Operation(allowExtraSuspension = true, blocking = true)
     suspend fun receiveViaSelect(): Any = try {
         select<Int> { c.onReceive { it } }
     } catch (e: NumberedCancellationException) {
         e.testResult
     }
 
-    @Operation(causesBlocking = true)
+    @Operation(causesBlocking = true, blocking = true)
     fun close(@Param(name = "closeToken") token: Int): Boolean = c.close(NumberedCancellationException(token))
 
-    // TODO: this operation should be (and can be!) linearizable, but is not
-    // @Operation
+    @Operation(causesBlocking = true, blocking = true)
     fun cancel(@Param(name = "closeToken") token: Int) = c.cancel(NumberedCancellationException(token))
 
-    // @Operation
-    fun isClosedForReceive() = c.isClosedForReceive
+    // @Operation TODO non-linearizable in BufferedChannel
+    open fun isClosedForReceive() = c.isClosedForReceive
 
-    // @Operation
+    @Operation(blocking = true)
     fun isClosedForSend() = c.isClosedForSend
 
-    // TODO: this operation should be (and can be!) linearizable, but is not
-    // @Operation
-    fun isEmpty() = c.isEmpty
+    // @Operation TODO non-linearizable in BufferedChannel
```

Comment:
It is the case, and it will be the case in this implementation. I believe that it is better to keep a comment in tests (I can omit TODOs, they are just nice to highlight the information).

Reply:
Sure, works for me